File: app/src/main/java/com/boost/espressotest/data/local/AppDatabase.java
package com.boost.espressotest.data.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

import com.boost.espressotest.data.local.dao.ProductDao;
import com.boost.espressotest.data.local.model.ProductContent;

/**
 * @author PerSpiKyliaTor on 22.01.18.
 */

@Database(entities = {ProductContent.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    public static final String APP_DATABASE_NAME = "app-database";

    public abstract ProductDao productDao();
}

File: app/src/main/java/com/boost/espressotest/data/local/AppDatabase.kt
package com.boost.espressotest.data.local

import androidx.room.Database
import androidx.room.RoomDatabase
import com.boost.espressotest.data.local.dao.ProductDao
import com.boost.espressotest.data.local.model.ProductEntity

@Database(entities = [ProductEntity::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun productDao(): ProductDao
}
